lua_modules/http/httpserver.lua

```diff
@@ -164,14 +174,11 @@ do
             ondata(connection, buf)
             -- buffer no longer needed
             buf = nil
-            -- NB: we explicitly reassign receive handler so that
-            --   next received chunks go directly to body handler
-            connection:on("receive", ondata)
```

Comment:
Hm.  In light of the change to `ondata`, it's going to close over `connection` rather than take it as an argument.  If you leave this chunk alone but revert `ondata` to take `connection` as an argument, do things remain copacetic?